src/Appwrite/Platform/Modules/Storage/Http/Tokens/Buckets/Files/Create.php

```diff
@@ -0,0 +1,115 @@
+        ->desc('Create file token')
+        ->groups(['api', 'token'])
+        ->label('scope', 'tokens.write')
+        ->label('event', 'tokens.[tokenId].create')
+        ->label('audits.event', 'token.create')
+        ->label('audits.resource', 'token/{response.$id}')
+        ->label('usage.metric', 'tokens.{scope}.requests.create')
+        ->label('usage.params', ['resourceId:{request.resourceId}', 'resourceType:{request.resourceType}'])
+        ->label('abuse-key', 'ip:{ip},method:{method},url:{url},userId:{userId}')
+        ->label('abuse-limit', APP_LIMIT_WRITE_RATE_DEFAULT)
+        ->label('abuse-time', APP_LIMIT_WRITE_RATE_PERIOD_DEFAULT)
+        ->label('sdk', new Method(
+            namespace: 'tokens',
+            name: 'createFileToken',
+            description: <<<EOT
+            Create a new token. A token is linked to a file or a bucket and manages permissions for those file(s). Token can be passed as a header or request get parameter.
+            EOT,
+            auth: [AuthType::SESSION, AuthType::KEY, AuthType::JWT],
+            responses: [
+                new SDKResponse(
+                    code: Response::STATUS_CODE_CREATED,
+                    model: Response::MODEL_RESOURCE_TOKEN,
+                )
+            ],
+            contentType: ContentType::JSON
+        ))
+        ->param('bucketId', '', new UID(), 'Storage bucket unique ID. You can create a new storage bucket using the Storage service [server integration](https://appwrite.io/docs/server/storage#createBucket).')
+        ->param('fileId', '', new UID(), 'File unique ID.')
+        ->param('expire', null, new Nullable(new DatetimeValidator()), 'Token expiry date', true)
+        ->param('permissions', [], new Permissions(APP_LIMIT_ARRAY_PARAMS_SIZE, [Database::PERMISSION_READ, Database::PERMISSION_UPDATE, Database::PERMISSION_DELETE, Database::PERMISSION_WRITE]), 'An array of permission strings. By default, only the current user is granted all permissions. [Learn more about permissions](https://appwrite.io/docs/permissions).', true)
+        ->inject('response')
+        ->inject('dbForProject')
+        ->inject('user')
+        ->inject('queueForEvents')
+        ->callback([$this, 'action']);
+    }
+
+    public function action(string $bucketId, string $fileId, ?string $expire, ?array $permissions, Response $response, Database $dbForProject, Document $user, Event $queueForEvents)
+    {
+
+        /**
+         * @var Document $bucket
+         * @var Document $file
+         */
+        ['bucket' => $bucket, 'file' => $file] = $this->getFileAndBucket($dbForProject, $bucketId, $fileId);
+
+        $fileSecurity = $bucket->getAttribute('fileSecurity', false);
+        $validator = new Authorization(Database::PERMISSION_UPDATE);
+        $bucketPermission = $validator->isValid($bucket->getUpdate());
+        if (!$fileSecurity && !$bucketPermission) {
+            throw new Exception(Exception::USER_UNAUTHORIZED);
+        }
```

Comment:
we could have an `if` block for just `!$fileSecurity`, that we we dont need to create a validator and use it when its not needed.